### ISSUE TITLE
train.fit epoch1 uses -inf

### ISSUE THREAD

**Issue by PGTBoos** [keras-team/keras#11703]
There seems to be an issue with model.fit in relation to loading previous weights.
The problems basically as seen in the logging  :
 
Epoch 00001: val_acc improved from -inf to 0.95089, saving model to \Chicken-weights.best.hdf5

Every fit starts with "-inf",  negative infinitive thus ignoring the last loaded weights results from previous runs. And so that will overwrite any best trainings from previous runs. if epoch 1 is worse it would still overwrite past saved results. due to "-inf", fit cannot improve from past runs.

```
checkpoint = ModelCheckpoint(filepath, monitor='val_acc', verbose=1, save_best_only=True, mode='max')
callbacks_list = [checkpoint]
model.fit(x_train, y_train, validation_data=(x_val, y_val), epochs=ep, batch_size=32, callbacks=callbacks_list, verbose=2)
```

**Comment by gabrieldemarmiesse** [comment]
I believe the issue is complicated since the `History` object is not saved with the model. We would need to rerun the model on the validation set before starting the training again, which might not be desirable by some users.
Plus the model has no knowledge that it was loaded rather than trained directly.

A simple fix would be to change `filepath` in your example to avoid overwriting. Supporting your use case would be extremely difficult.

**Comment by PGTBoos** [comment]
You got me wrong, the only problem i see here, that it starts with -inf
It's not the history i'm after, (in fact wouldn't want that), for distributed computing, and early stopping I need to be able to tell what the last best result was and not let it start all over again as it currently does. (not overwriting previous hdf5 with bad results !!). Having an options to give it a starting score to test against is all that is missing currently. 
In my view it wouldnt be hard to code either, (to pass a value for -inf instead) but i'm not owner of this source code.

**Comment by PGTBoos** [comment]
So please reopen this
